jsdoc/processors/check-access-tags.js

```diff
@@ -0,0 +1,44 @@
+module.exports = function checkAccessTagsProcessor(createDocMessage) {
+  return {
+    $runAfter: ['tags-extracted'],
+    $runBefore: ['processing-docs'],
+    $validate: { accessTypes: { presence: true } },
+    accessTypes: ['private', 'protected', 'public'],
+    $process: function(docs) {
+      var accessTypes = this.accessTypes;
+
+      docs.forEach(function(doc) {
+
+        // Check for invalid access type
+        if (doc.access && accessTypes.indexOf(doc.access) === -1) {
+          throw new Error(createDocMessage('"' + doc.access + '" is not an allowed access type for the "@access" tag. Try adding it to the checkAccessTagsProcessor.accessTypes array', doc));
+        }
+
+        // Check for multiple tags
+        var accessTypesFound = [];
+        accessTypes.forEach(function(accessType) {
+          if (doc[accessType]) {
+            accessTypesFound.push('@' + accessType);
+          }
+        });
+        if (doc.access) {
+          accessTypesFound.push('@access ' + doc.access);
+        }
+        if (accessTypesFound.length > 1) {
+          var tagList = accessTypesFound.map(function(tag) {
+            return '* ' + tag + '\n';
+          }).join('');
+          throw new Error(createDocMessage(
+            'Only one access type tag is allowed per document. Tags found were:\n' + tagList, doc));
+        }
+
+        // Convert "access" properties to explicit properties
+        if (doc.access) {
+          doc[doc.access] = true;
```

Comment:
I see that this could be problematic, if you managed to:

a) choose a name for an access type that was a property that already existed on the doc
b) explicitly added that name to the "standard" access types that are expected to be copied to booleans

It is also true that if I add a `@fileInfo` tag that wrote itself to the same property on the doc then that would make a mess of the whole doc generation process.

It is true though that the `@private` tag originally set itself as a boolean on the doc, which is one of the reasons why I felt we needed to refactor this feature.

Let me see if there is a more acceptable approach for you.